CMakeLists.txt

```diff
@@ -13,6 +13,8 @@ TRIBITS_PACKAGE_DECL(DataTransferKit)
 TRIBITS_ADD_DEBUG_OPTION()
 TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+
```

Comment:
This will affect everybody, right? Including user? I would prefer if we would only use the flag for jenkins. Also does `-Werror` work for every compiler?